src/ostorlab/apis/runners/authenticated_runner.py

```diff
@@ -37,11 +35,12 @@ class AuthenticatedAPIRunner(runner.APIRunner):
     """
 
     def __init__(self,
-                 username: str = None,
+                 username: str = None,  # Username and password to use the token based authentication
                  password: str = None,
-                 token_duration: str = None,
-                 proxy: str = None,
-                 verify: bool = True
+                 token_duration: str = None,  # Set a duration for the token validity
+                 proxy: str = None,  # The proxy through which a request is made
+                 verify: bool = True,  # Whether or not to verify the TLS certificate
+                 api_key: str = None  # Use API KEY based authentication. Used if token is not defined.
                  ):
         """Constructs all the necessary attributes for the object.
```

Comment:
Why is the docstring added this way, just use the standard google docstring way.